tubular/numeric.py

```diff
@@ -518,3 +525,190 @@ def transform(self, X):
+                0 < n_components < min(X.shape)
+            If randomized :
+                run randomized SVD by the method of Halko et al.
+            .. sklearn versionadded:: 0.18.0
+
+        random_state : int, RandomState instance or None, default=None
+            Used when the 'arpack' or 'randomized' solvers are used. Pass an int
+            for reproducible results across multiple function calls.
+            .. sklearn versionadded:: 0.18.0
+    Attributes
+    ----------
+
+    pca : PCA class from sklearn.decomposition
+    n_components_ : int
+        The estimated number of components. When n_components is set
+        to 'mle' or a number between 0 and 1 (with svd_solver == 'full') this
+        number is estimated from input data. Otherwise it equals the parameter
+        n_components, or the lesser value of n_features and n_samples
+        if n_components is None.
+    feature_names_out: list or None
+        list of feature name representing the new dimensions.
+
+
+    """
+
+    def __init__(
+        self, columns, n_components=2, svd_solver="auto", random_state=None, **kwargs
+    ):
+
+        super().__init__(columns=columns, **kwargs)
+
+        if type(n_components) is int:
+            if n_components < 1:
+                raise ValueError(
+                    f"{self.classname()}:n_components must be positive  got {str(n_components)}"
+                )
+            else:
+                self.n_components = n_components
+        else:
+            if n_components == "mle":
+                self.n_components = n_components
+            else:
+                raise TypeError(
+                    f"{self.classname()}:unexpected type {type(n_components)} for n_components, must be int or equal to 'mle'."
+                )
+
+        if type(svd_solver) is str:
+            if svd_solver not in ["auto", "full", "arpack", "randomized"]:
+                raise ValueError(
+                    f"{self.classname()}:svd_solver {svd_solver} is unknown. Please select among 'auto', 'full', 'arpack', 'randomized'."
+                )
+            else:
+                self.svd_solver = svd_solver
+        else:
+            raise TypeError(
+                f"{self.classname()}:unexpected type {type(svd_solver)} for svd_solver, must be str"
+            )
+
+        if type(random_state) is int:
+            self.random_state = random_state
+        else:
+            if random_state is None:
+                self.random_state = random_state
+            else:
+                raise TypeError(
+                    f"{self.classname()}:unexpected type {type(random_state)} for random_state, must be int or None."
+                )
+
+        self.pca = PCA(
+            n_components=self.n_components,
+            svd_solver=self.svd_solver,
+            random_state=self.random_state,
+        )
+        self.feature_names_out = None
+        self.n_components_ = None
+
+    def check_numeric_columns(self, X):
```

Comment:
This method would be useful in other numeric transformers - if you have time please consider moving it to a BaseNumericTransformer class so it can be reused.  If not we can refactor this in a later release.

Reply:
related to a new git issue check_numeric_columns ?